### PR TITLE
Crypto-18: Integrate Intro Screen logic

### PR DESCRIPTION
**Pull request by vitalyaliskevych**
### Whats News
- Integrate Intro Screen logic
### Issues
- [Crypto-18](https://trello.com/c/bZGr2gt4)
### Video

https://user-images.githubusercontent.com/98754902/220095657-8a67d8d8-8d16-4f97-8c2b-eb7ea7c65948.mov

### CODE REVIEW

File: Crypto/Flows/Coordinator/IntroCoordinator/IntroCoordinator.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+//  Created by Developer on 20.02.2023.
+//
+
+import Foundation
+import SwiftUI
+
+class IntroCoordinator: ObservableObject {
+    
+    enum Route {
+        case onMainSelected(coordinator: MainCoordinator)
```
Comment (romansorochak) [inline Crypto/Flows/Coordinator/IntroCoordinator/IntroCoordinator.swift:14]: just `main`
This is not an action or event. This is a route. And here you actually know that the route is `main`

File: Crypto/Flows/Coordinator/IntroCoordinator/IntroCoordinator.swift [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    
+    @Published var route: Route?
+    
+    var introViewModel: IntroViewModel
+    
+    init(introViewModel: IntroViewModel) {
+        self.introViewModel = introViewModel
+        introViewModel.onResult = { [weak self] result in
+            switch result {
+            case .onMainSelected(let coinModels):
```
Comment (romansorochak) [inline Crypto/Flows/Coordinator/IntroCoordinator/IntroCoordinator.swift:25]: Though here it is make sense to have case `onMainSelected`.
As user does something on `View`. Then `ViewModel` procedures user's action and returns result - `onMainSelected`. Then here in Coordinator you actually decide where to navigate user to. So inside `ViewModel` you deal with results (you don't know anything about the place to where we will navigate). On the other hand `Coordinator` knows and actually decides to which place we should navigate to.

File: Crypto/Flows/Coordinator/IntroCoordinator/IntroCoordinatorView.swift [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    var body: some View {
+        NavigationView {
+            ZStack {
+                IntroView(viewModel: coordinator.introViewModel)
+                NavigationLink(
+                    unwrapping: $coordinator.route,
+                    case: /IntroCoordinator.Route.onMainSelected,
+                    destination: {(coordinator: Binding<MainCoordinator>) in
+                        MainCoordinatorView(
+                            coordinator: coordinator.wrappedValue).navigationBarBackButtonHidden(true)
```
Comment (romansorochak) [inline Crypto/Flows/Coordinator/IntroCoordinator/IntroCoordinatorView.swift:24]: move `).navigationBarBackButtonHidden(true)` to the next line

File: Crypto/Flows/Coordinator/IntroCoordinator/IntroCoordinatorView.swift [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+struct IntroCoordinatorView: View {
+    
+    @ObservedObject var coordinator: IntroCoordinator
+    
+    var body: some View {
+        NavigationView {
+            ZStack {
+                IntroView(viewModel: coordinator.introViewModel)
+                NavigationLink(
```
Comment (romansorochak) [inline Crypto/Flows/Coordinator/IntroCoordinator/IntroCoordinatorView.swift:19]: that is not good
What is a reason of keeping `IntroView` in navigation stack.
Instead just replace `IntroView` with `MainCoordinatorView`.

And actually that needs to be done in `AppCoordinator`.
Also I don't see a point to have `IntroCoordinator` at all. 
You can have `AppCoordinator` that can show either `IntroView` either `MainCoordinatorView`. 
But you still should not use `NavigationLink` here. Instead use just simple switch case between two views.

File: Crypto/Flows/Coordinator/IntroCoordinator/IntroCoordinatorView.swift [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+                    unwrapping: $coordinator.route,
+                    case: /IntroCoordinator.Route.onMainSelected,
+                    destination: {(coordinator: Binding<MainCoordinator>) in
```
Comment (vasyl-nadtochii) [inline Crypto/Flows/Coordinator/IntroCoordinator/IntroCoordinatorView.swift:22]: change it to `destination: { coordinator in`

File: Crypto/Flows/Coordinator/IntroCoordinator/IntroCoordinatorView.swift [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+                        MainCoordinatorView(
+                            coordinator: coordinator.wrappedValue).navigationBarBackButtonHidden(true)
+                    }, onNavigate: {_ in}) {}
```
Comment (vasyl-nadtochii) [inline Crypto/Flows/Coordinator/IntroCoordinator/IntroCoordinatorView.swift:25]: move `) {}` to next line

File: Crypto/Flows/Screens/IntroScreen/IntroViewModel.swift [hunk 7]
```diff
@@ -6,3 +6,21 @@
...
 import Foundation
+import SwiftUI
+
+class IntroViewModel: ObservableObject {
+    
+    enum Result {
+        case onMainSelected(coinModels: Coin)
+    }
+    
+    @Published var animation = false
```
Comment (vasyl-nadtochii) [inline Crypto/Flows/Screens/IntroScreen/IntroViewModel.swift:17]: `titleIsShown` fits better

File: Crypto/Flows/Screens/IntroScreen/IntroViewModel.swift [hunk 8]
```diff
@@ -6,3 +6,21 @@
...
+    @Published var animationButton = false
```
Comment (vasyl-nadtochii) [inline Crypto/Flows/Screens/IntroScreen/IntroViewModel.swift:18]: `buttonIsShown`

File: Crypto/Flows/Screens/IntroScreen/IntroViewModel.swift [hunk 9]
```diff
@@ -6,3 +6,21 @@
...
+    
+    var onResult: ((Result) -> Void)?
+    var coin = Coin(name: "", price: 1.2, icon: Image.btcIcn)
+    
+    func selectMainView(coinModels: Coin) {
```
Comment (vasyl-nadtochii) [inline Crypto/Flows/Screens/IntroScreen/IntroViewModel.swift:23]: better to name it `onStartButtonPressed`

**Review by vasyl-nadtochii — COMMENTED** [review]
1. Resolve comments
2. From your video I can see that button and text appearing are not animated. Please handle that
3. Make Intro background darker (check design)

**Comment by vitalyaliskevych** [comment]
I have bug with toolbar. I crated BugTicked and fix this

**Pull request merged**